identity-service/src/utils/contentNodeIPCheckTest.js

```diff
@@ -0,0 +1,69 @@
+const assert = require('assert')
+const sinon = require('sinon')
+const apiSigning = require('./apiSigning')
+const { _isIPFromContentNode } = require('./contentNodeIPCheck')
+
+describe('test contentNodeIPCheck', () => {
+  it('returns true if already known', () => {
+    const isIPFromContentNode = _isIPFromContentNode(
+      '127.0.0.1',
+      {
+        body: {
+          signature: '0x16092e63f925bc0bd2829348deadf4a3404cdb1067669fcad5bea391e54059285c292a8bad5360aca82abb9b72d0f513ccba48c99b6f99f4c4559cff7b0032251c',
+          timestamp: '2021-03-16T07:25:10.351Z'
+        }
+      },
+      new Set([]),
+      new Set(['127.0.0.1'])
+    )
+    assert.strictEqual(isIPFromContentNode, true)
+  })
+
+  it('returns false if missing signature', () => {
+    const isIPFromContentNode = _isIPFromContentNode(
+      '127.0.0.1',
+      {
+        body: {
+          timestamp: '2021-03-16T07:25:10.351Z'
+        }
+      },
+      new Set([]),
+      new Set([])
+    )
+    assert.strictEqual(isIPFromContentNode, false)
+  })
+
+  it('returns false if signature is expired', () => {
+    const isIPFromContentNode = _isIPFromContentNode(
+      '127.0.0.1',
+      {
+        body: {
+          signature: '0x16092e63f925bc0bd2829348deadf4a3404cdb1067669fcad5bea391e54059285c292a8bad5360aca82abb9b72d0f513ccba48c99b6f99f4c4559cff7b0032251c',
+          timestamp: '2021-03-01T07:25:10.351Z'
+        }
+      },
+      new Set([]),
+      new Set([])
+    )
+    assert.strictEqual(isIPFromContentNode, false)
+  })
+
+  it('returns true if signature produces a valid known wallet', () => {
+    // 0xabc is a valid known content node
+    sinon.stub(apiSigning, 'recoverWallet').returns('0xabc')
+
+    const isIPFromContentNode = _isIPFromContentNode(
+      '127.0.0.1',
+      {
+        body: {
+          signature: '0x16092e63f925bc0bd2829348deadf4a3404cdb1067669fcad5bea391e54059285c292a8bad5360aca82abb9b72d0f513ccba48c99b6f99f4c4559cff7b0032251c',
```

Comment:
since this signature should have the timestamp in it, how does this work with a hardcoded signature?

Reply:
because like a fool i was checking the wrong timestamp (the one they passed in vs. the one in the signature)

Reply:
this test case is just wrong. fixing :)

Reply:
without modifying contentNodeIPCheck.js this test is now correct. we can still check for expiry based on the body arg so we can bypass decoding if necessary and if the body.timestamp arg doesn't match what's encoded into the signature, the recovery will fail to produce the correct results